pkg/instancegroups/instancegroups.go

```diff
@@ -624,6 +625,38 @@ func (c *RollingUpdateCluster) drainNode(u *cloudinstances.CloudInstance) error
 		return fmt.Errorf("node name not set")
 	}
 
+	hasAPIServer := u.CloudInstanceGroup.InstanceGroup.HasAPIServer()
+
+	if hasAPIServer {
+		pods, err := c.K8sClient.CoreV1().Pods(metav1.NamespaceSystem).List(context.Background(), metav1.ListOptions{
+			FieldSelector: fields.SelectorFromSet(
+				fields.Set{"spec.nodeName": u.Node.Name},
+			).String(),
+		})
+
+		if err != nil {
+			return fmt.Errorf("error listing pods on node %q: %v", u.Node.Name, err)
+		}
+
+		for _, pod := range pods.Items {
+			if pod.Annotations["dns.alpha.kubernetes.io/internal"] != "" {
+
+				patchBody := []byte(`{"metadata":{"annotations":{"dns.alpha.kubernetes.io/internal":null}}}`)
+
+				_, err := c.K8sClient.CoreV1().
+					Pods(metav1.NamespaceSystem).
+					Patch(context.Background(), pod.Name, types.StrategicMergePatchType, patchBody, metav1.PatchOptions{})
+
+				if err != nil {
+					return fmt.Errorf("error removing dns.alpha.kubernetes.io/internal annotation from pod %q: %v", pod.Name, err)
+				}
+			}
+		}
+
+		// Sleep for twice the default DNS record TTL to ensure that downstream clients have refreshed their DNS caches before we delete the node.
+		time.Sleep(2 * time.Minute)
```

Comment:
Some operators may use a load balancer instead of DNS, others may use external-dns and probably there are other possibilities.
```suggestion
		if len(pods.Items) > 0 {
			// Sleep for twice the default DNS record TTL to ensure that downstream clients have refreshed their DNS caches before we delete the node.
			time.Sleep(2 * time.Minute)
		}
```

Reply:
Hey @hakman, thanks for your suggestion! I've added a check to skip the sleep if we haven't removed the internal DNS annotation from any pods.